File: typescript/src/watch.ts
import { Kernel } from "./core.js";
import { discover } from "./discovery.js";

console.log("Use nodemon, tsx watch, or your runtime watcher to restart on file changes.");

const { modules, adapters } = discover();
const kernel = new Kernel(modules, adapters);
await kernel.start();
kernel.ctx.resolve<{ run(): void }>("app", [1, 0]).run();
await kernel.stop();

File: watch.py
from lattice import HotReloadManager, Kernel, discover


def main():
    modules, adapters = discover("plugins")
    kernel = Kernel(modules, adapters)
    kernel.start()

    app = kernel.ctx.resolve("app", (1, 0))
    app.run()

    manager = HotReloadManager(kernel, "plugins", "plugins")
    manager.watch(interval_seconds=1.0)


if __name__ == "__main__":
    main()
